spec/integration/bad2_spec.lua

```diff
@@ -20,6 +20,7 @@ local snowplow
 describe("Integration tests with HTTP/collector problems", function()
   setup(function()
     _G._TEST = true
+    _G.os = require("spec/mocks/mock_os")
```

Comment:
This is kind of a theoretical point and doesn't really make a difference in practice, but I would argue that integration tests should not mock stuff. They should be as close to reality as possible. I think it would be slightly better to let the tracker call the actual implementation of `os.time()` and check that it assigned some reasonable number. It would complicate your life as you wouldn't be able to directly compare the strings. On the other hand, it opens an opportunity to overengineer the tests and make use of [Lua patterns](https://www.lua.org/pil/20.2.html). I'm fine with leaving it like this, but if you want to take this opportunity to overengineer stuff, it would theoretically make the integration tests a bit more robust.

Reply:
Btw, I just looked in the good1 and bad1 integration tests and it seems that the timestamp is [still passed there](https://github.com/snowplow/snowplow-lua-tracker/blob/issue/33-remove_user_overridable_dtm/spec/integration/good1_spec.lua#L29). Interesting that the test passes, I guess the arguments are ignored. Just a bit confusing for someone reading it in the future.

Reply:
Trying to escape the URL for pattern matching turned out to be a bit of a nightmare, so I came up with an alternate solution of spying on the `os.time` function and using the most recent return val from it to construct the expected string. (also looks a bit cleaner and more understandable to anyone in the future looking at the test IMO)

The timestamps in good1 and bad1 have also been removed.

Reply:
Interesting approach! Looks good!